network-store-server/src/main/java/com/powsybl/network/store/server/LimitsHandler.java

```diff
@@ -0,0 +1,358 @@
+import static com.powsybl.network.store.server.QueryCatalog.SELECTED_OPERATIONAL_LIMITS_GROUP_ID2;
+
+/**
+ * @author Etienne Lesot <etienne.lesot at rte-france.com>
+ */
+@Component
+public class LimitsHandler {
+    private final DataSource dataSource;
+    private final ObjectMapper mapper;
+
+    public LimitsHandler(DataSource dataSource, ObjectMapper mapper) {
+        this.dataSource = dataSource;
+        this.mapper = mapper;
+    }
+
+    public Optional<LimitsInfos> getLimitsInfos(UUID networkUuid, int variantNum, String type, String equipmentId) {
+        List<TemporaryLimitAttributes> temporaryLimitAttributes = getTemporaryLimits(networkUuid, variantNum, type, equipmentId);
+        List<PermanentLimitAttributes> permanentLimitAttributes = getPermanentLimits(networkUuid, variantNum, type, equipmentId);
+        if (permanentLimitAttributes.isEmpty() && temporaryLimitAttributes.isEmpty()) {
+            return Optional.empty();
+        }
+        return Optional.of(new LimitsInfos(permanentLimitAttributes, temporaryLimitAttributes));
+    }
+
+    // temporary limits
+    public List<TemporaryLimitAttributes> getTemporaryLimits(UUID networkUuid, int variantNum, String type, String equipmentId) {
+        try (Connection connection = dataSource.getConnection()) {
+            var preparedStmt = connection.prepareStatement(QueryCatalog.buildTemporaryLimitQuery());
+            preparedStmt.setString(1, networkUuid.toString());
+            preparedStmt.setInt(2, variantNum);
+            preparedStmt.setString(3, type);
+            preparedStmt.setString(4, equipmentId);
+
+            return innerGetTemporaryLimits(preparedStmt);
+        } catch (SQLException e) {
+            throw new UncheckedSqlException(e);
+        }
+    }
+
+    private List<TemporaryLimitAttributes> innerGetTemporaryLimits(PreparedStatement preparedStmt) throws SQLException {
+        try (ResultSet resultSet = preparedStmt.executeQuery()) {
+            if (resultSet.next()) {
+                String temporaryLimitData = resultSet.getString(1);
+                List<TemporaryLimitSqlData> parsedTemporaryLimitData = mapper.readValue(temporaryLimitData, new TypeReference<>() { });
+                return parsedTemporaryLimitData.stream().map(TemporaryLimitSqlData::toTemporaryLimitAttributes).toList();
+            }
+            return Collections.emptyList();
+        } catch (JsonProcessingException e) {
+            throw new UncheckedIOException(e);
+        }
+    }
+
+    // permanent limits
+    public List<PermanentLimitAttributes> getPermanentLimits(UUID networkUuid, int variantNum, String type, String equipmentId) {
+        try (Connection connection = dataSource.getConnection()) {
+            var preparedStmt = connection.prepareStatement(QueryCatalog.buildPermanentLimitQuery());
+            preparedStmt.setString(1, networkUuid.toString());
+            preparedStmt.setInt(2, variantNum);
```

Comment:
Your code should manage partial variants.
See https://github.com/powsybl/powsybl-network-store-server/blob/01f6e753171200bfc2f47741fef7ec78bd48e509/network-store-server/src/main/java/com/powsybl/network/store/server/ExtensionHandler.java#L221